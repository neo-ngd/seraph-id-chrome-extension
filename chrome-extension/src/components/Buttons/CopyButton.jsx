// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

import React from 'react';
import PropTypes from 'prop-types';
import { useClipboard } from 'use-clipboard-copy';
import { Button, Tooltip } from '@material-ui/core';
import dictionary from "../../commons/dictionary";

/**
 * <CopyButton />
 * The button is copying passing text to the clipboard.
 * @param textToCopy
 * @param children
 * @return {*}
 * @constructor
 */
const CopyButton = ({ textToCopy, children }) => {
  const clipboard = useClipboard({
    copiedTimeout: 700, // timeout duration in milliseconds
  });

  return (
    <Tooltip title={clipboard.copied ? dictionary.copyButton.copied : dictionary.copyButton.copy}>
      <Button
        variant="text"
        onClick={() => {
          clipboard.copy(textToCopy);
        }}
      >
        {children}
      </Button>
    </Tooltip>
  );
};

CopyButton.propTypes = {
    textToCopy: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
};

export default CopyButton;
