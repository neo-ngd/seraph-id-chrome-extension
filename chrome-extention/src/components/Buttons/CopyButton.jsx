import React from 'react';
import PropTypes from 'prop-types';
import { useClipboard } from 'use-clipboard-copy';
import { Button, Tooltip } from '@material-ui/core';
import dictionary from "../../commons/dictionary";

const CopyButton = ({ textToCopy, children }) => {
  const clipboard = useClipboard({
    copiedTimeout: 700, // timeout duration in milliseconds
  });

  return (
    <Tooltip title={clipboard.copied ? dictionary.copyButton.copied : dictionary.copyButton.copy}>
      <Button
        variant="text"
        onClick={() => {
          clipboard.copy(textToCopy);
        }}
      >
        {children}
      </Button>
    </Tooltip>
  );
};

CopyButton.propTypes = {
    textToCopy: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
};

export default CopyButton;
