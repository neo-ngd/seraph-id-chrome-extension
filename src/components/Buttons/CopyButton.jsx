import React from 'react';
import { useClipboard } from 'use-clipboard-copy';
import { Button, Tooltip, Typography } from '@material-ui/core/';

export default function PublicUrl({ textToCopy, children }) {
  const clipboard = useClipboard({
    copiedTimeout: 700, // timeout duration in milliseconds
  });

  return (
    <Tooltip title={clipboard.copied ? 'Copied!' : 'Copy Link'}>
      <Button
        variant="text"
        onClick={() => {
          clipboard.copy(textToCopy);
        }}
      >
        {children}
      </Button>
    </Tooltip>
  );
}
