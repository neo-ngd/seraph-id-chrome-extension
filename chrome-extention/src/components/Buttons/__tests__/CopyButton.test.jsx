import React from 'react';
import { shallow } from 'enzyme';
import CopyButton from "../CopyButton";

const props = {
    textToCopy: 'test',
    children: 'test-children',
};

describe('<CopyButton />', () => {
    it('should render component', () => {
        const wrapper = shallow(<CopyButton {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
