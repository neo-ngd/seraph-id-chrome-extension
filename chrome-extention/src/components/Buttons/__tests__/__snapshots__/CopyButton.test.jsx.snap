// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyButton /> should render component 1`] = `
<WithStyles(ForwardRef(Tooltip))
  title="Copy address!"
>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    variant="text"
  >
    test-children
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(Tooltip))>
`;
